Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -2,29 +2,95 @@
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+            var currentTimeLowerBounds = new DateTime[exactPostTime.Length];
+            var currentTimeUpperBounds = new DateTime[exactPostTime.Length];
+
+            for (int i = 0; i < exactPostTime.Length; i++)
+            {
+                var exactTime = DateTime.Parse(exactPostTime[i], System.Globalization.CultureInfo.InvariantCulture);
+                var currentTimeLowerBound = exactTime;
+                var currentTimeUpperBound = exactTime;
+                var displayedPostTimeComponents = showPostTime[i].Split(' ');
+
+                //Determining possible corresponding upper bound and lower vound values of current time.
+                switch (displayedPostTimeComponents[1])
+                {
+                    case "seconds":
+                        currentTimeUpperBound = exactTime.AddSeconds(59);
+                        break;
+
+                    case "minutes":
+                        currentTimeLowerBound = exactTime.AddMinutes(double.Parse(displayedPostTimeComponents[0]));
+                        currentTimeUpperBound = currentTimeLowerBound.AddSeconds(59);
+                        break;
+
+                    case "hours":
+                        currentTimeLowerBound = exactTime.AddHours(double.Parse(displayedPostTimeComponents[0]));
+                        currentTimeUpperBound = currentTimeLowerBound.AddMinutes(59);
+                        currentTimeUpperBound = currentTimeLowerBound.AddSeconds(59);
+                        break;
+
+                    default:
+                        break;
+                }
+
+                currentTimeLowerBounds[i] = currentTimeLowerBound;
+                currentTimeUpperBounds[i] = currentTimeUpperBound;
+            }
+
+            for (int i = 0; i < exactPostTime.Length - 1; i++)
+            {
+                for (int j = i + 1; j < exactPostTime.Length; j++)
+                {
+                    if (exactPostTime[i] == exactPostTime[j])
+                    {
+                        if (showPostTime[i] != showPostTime[j])
+                        {
+                            return "impossible";
+                        }
+                    }
+                }
+            }
+
+            //sorting upperbounds in ascending order and lower bounds in descending order
+            Array.Sort(currentTimeUpperBounds);
+            Array.Sort(currentTimeLowerBounds);
+            Array.Reverse(currentTimeLowerBounds);

```

Comment:
It's one way to sort an array in descending order. However, you can explore the other ways to do that in a single operation like using CompareTo or delegate approach. You can find such on geeksforGeeks website.

Reply:
Applied CompareTo approach.